tokio/src/sync/watch.rs

```diff
@@ -10,24 +10,64 @@
 //!
 //! [`channel`] returns a [`Sender`] / [`Receiver`] pair. These are the producer
 //! and consumer halves of the channel. The channel is created with an initial
-//! value. The **latest** value stored in the channel is accessed with
-//! [`Receiver::borrow()`]. Awaiting [`Receiver::changed()`] waits for a new
-//! value to be sent by the [`Sender`] half.
+//! value.
+//!
+//! Each [`Receiver`] independently tracks the last value *seen* by its caller.
+//!
+//! To access the **latest** value stored in the channel and mark it as *seen*,
+//! use [`Receiver::borrow_and_update()`].
+//!
+//! To access the latest value **without** changing its state to *seen*, use
+//! [`Receiver::borrow()`]. (If the value has already been marked *seen*,
+//! [`Receiver::borrow()`] is equivalent to [`Receiver::borrow_and_update()`].)
```

Comment:
do you think it's worth explicitly stating something like "`borrow_and_update()` should be preferred if the receiver intends to await notifications from `changed()`, but it is not required if the receiver is just looking at the current value of the shared state without waiting for changes"?

Reply:
Added this to the new section, and also mentioned why `borrow` can be more convenient (it's `&self`).

Reply:
do you think we should maybe add a 
```suggestion
//! [`Receiver::borrow()`]. (If the value has already been marked *seen*,
//! [`Receiver::borrow()`] is equivalent to [`Receiver::borrow_and_update()`].)
//!
//! See [here](#borrow-and-update-versus-borrow) for more information on when
//! to use these methods.
```
or something?